src/DynamoUtilities/PathHelper.cs

```diff
@@ -61,7 +75,10 @@ public static bool IsReadOnlyPath(string filePath)
                 // We mark the path read only when
                 // 1. file read-only
                 // 2. user does not have write access to the folder
-                return Finfo.IsReadOnly || !HasWritePermissionOnDir(Finfo.Directory.ToString());
+
+                // We have no cross platform Directory access writes APIs.
+                bool hasWritePermissionOnDir = OSHelper.IsWindows() ? HasWritePermissionOnDir(Finfo.Directory.ToString()) : true;
```

Comment:
I guess now wherever this is used should be wrapped in try catch?
Maybe update summary?

Reply:
Not sure yet how we should go about it for linux. There are packages that have similar APIs ...(as far as I remember)
Or we can implement a simple version ourselves.

I think we need to do this in a separate task...talk about acceptance criteria and maybe some spikes too